Currently fails:
```
| ERROR: Execution of '/home/esc/ws/remarkable/codex-distro/repo/build/tmp/work/cortexa7hf-neon-remarkable-linux-gnueabi/qt-rappor-client/1.0+gitAUTOINC+db5d6ba75e-r0/temp/run.do_configure.376' failed with exit code 1:
| -- The CXX compiler identification is GNU 9.3.0
| -- The C compiler identification is GNU 9.3.0
| -- Check for working CXX compiler: /home/esc/ws/remarkable/codex-distro/repo/build/tmp/work/cortexa7hf-neon-remarkable-linux-gnueabi/qt-rappor-client/1.0+gitAUTOINC+db5d6ba75e-r0/recipe-sysroot-native/usr/bin/arm-remarkable-linux-gnueabi/arm-remarkable-linux-gnueabi-g++
| -- Check for working CXX compiler: /home/esc/ws/remarkable/codex-distro/repo/build/tmp/work/cortexa7hf-neon-remarkable-linux-gnueabi/qt-rappor-client/1.0+gitAUTOINC+db5d6ba75e-r0/recipe-sysroot-native/usr/bin/arm-remarkable-linux-gnueabi/arm-remarkable-linux-gnueabi-g++ -- works
| -- Detecting CXX compiler ABI info
| -- Detecting CXX compiler ABI info - done
| -- Detecting CXX compile features
| -- Detecting CXX compile features - done
| -- Check for working C compiler: /home/esc/ws/remarkable/codex-distro/repo/build/tmp/work/cortexa7hf-neon-remarkable-linux-gnueabi/qt-rappor-client/1.0+gitAUTOINC+db5d6ba75e-r0/recipe-sysroot-native/usr/bin/arm-remarkable-linux-gnueabi/arm-remarkable-linux-gnueabi-gcc
| -- Check for working C compiler: /home/esc/ws/remarkable/codex-distro/repo/build/tmp/work/cortexa7hf-neon-remarkable-linux-gnueabi/qt-rappor-client/1.0+gitAUTOINC+db5d6ba75e-r0/recipe-sysroot-native/usr/bin/arm-remarkable-linux-gnueabi/arm-remarkable-linux-gnueabi-gcc -- works
| -- Detecting C compiler ABI info
| -- Detecting C compiler ABI info - done
| -- Detecting C compile features
| -- Detecting C compile features - done
| CMake Error at CMakeLists.txt:27 (set_target_properties):
|   set_target_properties called with incorrect number of arguments.

```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,10 +1,12 @@
 cmake_minimum_required(VERSION 3.9)
-project(qt-rappor-client LANGUAGES CXX C)
+project(qt-rappor-client LANGUAGES CXX C VERSION 1.0.0)
 
 option(QT_RAPPOR_STATIC "Build static library so it can be built into the application")
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+include(GNUInstallDirs)
 
 find_package(Qt5 COMPONENTS Core REQUIRED)
 
@@ -22,7 +24,10 @@ if(QT_RAPPOR_STATIC)
     add_library(qt-rappor STATIC ${QT_RAPPOR_SRC})
 else()
     add_library(qt-rappor SHARED ${QT_RAPPOR_SRC})
-    set_property(TARGET qt-rappor PROPERTY POSITION_INDEPENDENT_CODE True)
+    set_target_properties(qt-rappor PROPERTIES
+        POSITION_INDEPENDENT_CODE True
+        VERSION ${PROJECT_VERSION}
+        SOVERSION ${PROJECT_VERSION_MAJOR})
 endif()
 target_link_libraries(qt-rappor Qt5::Core)
 target_include_directories(qt-rappor PUBLIC .)
@@ -55,7 +60,7 @@ else()
     message(STATUS "Skipping tests")
 endif()
 
-install(TARGETS qt-rappor DESTINATION lib)
+install(TARGETS qt-rappor LIBRARY DESTINATION lib)
 
 install(FILES qt-rappor-client/encoder.h DESTINATION include/qt-rappor-client)
 install(FILES qt-rappor-client/rappor_deps.h DESTINATION include/qt-rappor-client)

